src/auth/react-auth0-spa.tsx

```diff
@@ -73,6 +76,42 @@ export const Auth0Provider = ({
     setIsAuthenticated(true);
     setUser(user);
   };
+
+  useEffect(() => {
+    async function updateStoredUser() {
+      if (user !== undefined) {
+        const savedUser = await getUser(user.sub);
+        if (!savedUser) {
+          // this is expected, as auth0 users predate this feature.
+          // we can lazy-migrate them into our user database by
+          // creating a record now.
+          saveUser({
+            name: user.name,
+            email: user.email,
+            auth0Id: user.sub,
+          });
+        } else {
+          // if anything has changed, update our records.
+          // this anoints Auth0 as the ultimate source of truth
+          // for user info, synced at the beginning of a session
+          if (
+            !isEqualWith(
+              savedUser,
+              user,
+              (a, b) => a.name === b.name && a.email === b.email,
+            )
+          ) {
+            saveUser({
+              ...savedUser,
+              ...pick(user, ["email", "name"]),
+            });
+          }
+        }
+      }
+    }
+    updateStoredUser();
+  }, [user]);
```

Comment:
So, if I understand correctly, the net effect of this is that the hook is only called when the `user` object changes, which should only be when the user logs into a new session, right?

Reply:
it'll run every time we authenticate them, which is essentially on every page load (after login or immediately if they are already logged in)